feat(terraform): add support for private terraform registries (#4964)
bridgecrewio/checkov: docs/7.Scan Examples/Terraform.md

### Scanning Private Terraform Modules

> In case third-party modules are stored in a private repository or a private terraform registry (hosted on Terraform Cloud / Enterprise or a third-party provider like Gitlab), you can provide access tokens as environment variables for checkov to attempt to clone those modules.

Reviewer: ```suggestion
If you have modules stored in a private repository or a private Terraform registry (hosted on Terraform Cloud, Terraform Enterprise or a third-party provider like GitLab), you can grant Checkov access by providing access tokens as environment variables. This will enable Checkov to attempt to clone and scan those modules.
```

> In case third-party modules are stored in a private repository or a private terraform registry (hosted on Terraform Cloud / Enterprise or a third-party provider like Gitlab), you can provide access tokens as environment variables for checkov to attempt to clone those modules.

Reviewer: Glad that GitLab registry will work!

| Variable Name          | Description                                                                                      |
|------------------------|--------------------------------------------------------------------------------------------------|
| GITHUB_PAT             | Github personal access token with read access to the private repository                          |
| BITBUCKET_TOKEN        | Bitbucket personal access token with read access to the private repository                       |
| TF_HOST_NAME           | (defaults to app.terraform.io) Terraform Enterprise host name: example.com                       |
| TFC_TOKEN*             | (deprecated, use TF_REGISTRY_TOKEN) Terraform Cloud token which can access the private registry  |

> | TF_REGISTRY_TOKEN      | Private registry access token (supports terraform cloud / enterprise and third-party registries) |

Reviewer: ```suggestion
| TF_REGISTRY_TOKEN      | Private registry access token (supports Terraform Cloud / Enterprise and third-party registries) |
```

| VCS_BASE_URL  | Base URL of the self-hosted VCS: https://example.com |
| VCS_USERNAME  | Username for basic authentication                    |
| VCS_TOKEN     | Password for basic authentication                    |

> #### Examples

Reviewer 2: 100 points
